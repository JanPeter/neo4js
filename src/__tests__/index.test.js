// @flow

import trineo from "../index";

describe("trineo", () => {
  beforeAll(() => {
    trineo.init({
      boltUri: "localhost",
      boltPort: 10001,
    });
  });

  afterEach(async () => {
    await trineo.run("MATCH (n) DETACH DELETE n");
  });

  afterAll(() => {
    trineo.close();
  });

  it("should create a person", async () => {
    const result = await trineo.run(
      'CREATE (n:Person {name:"Hanns"}) RETURN n'
    );
    expect(result).toMatchSnapshot();
  });

  it("should create a bunch of persons and select them", async () => {
    await trineo.run(
      'CREATE (n:Person {name:"Hanns"}), (p:Person {name:"Huber"})'
    );
    const result = await trineo.run("MATCH (n:Person) RETURN n");
    expect(result).toMatchSnapshot();
  });

  it("should create a person with params", async () => {
    const result = await trineo.run("CREATE (n:Person {p}) RETURN n", {
      p: { name: "Hanns" },
    });
    expect(result).toMatchSnapshot();
  });

  it("should produce an error", async () => {
    try {
      await trineo.run("CREATE");
      expect(1).toEqual(0);
    } catch (err) {
      expect(err).toMatchSnapshot();
    }
  });
});
