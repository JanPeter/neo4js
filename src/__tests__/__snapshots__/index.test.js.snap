// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`neo4js should create a bunch of persons and select them 1`] = `
Array [
  Object {
    "n": Object {
      "name": "Hanns",
    },
  },
  Object {
    "n": Object {
      "name": "Huber",
    },
  },
]
`;

exports[`neo4js should create a person 1`] = `
Array [
  Object {
    "n": Object {
      "name": "Hanns",
    },
  },
]
`;

exports[`neo4js should create a person with params 1`] = `
Array [
  Object {
    "n": Object {
      "name": "Hanns",
    },
  },
]
`;

exports[`neo4js should produce an error 1`] = `
[Error: Unexpected end of input: expected whitespace, CONSTRAINT, INDEX, UNIQUE or a pattern (line 1, column 7 (offset: 6))
"CREATE"
       ^]
`;
